app/src/main/java/org/mozilla/reference/browser/NotificationManager.kt

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.reference.browser
+
+import android.annotation.TargetApi
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager as AndroidNotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.Build.VERSION.SDK_INT
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import mozilla.components.concept.sync.Device
+import mozilla.components.concept.sync.TabData
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Manages notification channels and allows displaying different types of notifications.
+ */
+class NotificationManager(private val context: Context) {
+    companion object {
+        const val RECEIVE_TABS_TAG = "org.mozilla.reference.browser.ReceivedTabs"
+        const val RECEIVE_TABS_CHANNEL_ID = "org.mozilla.reference.browser.ReceivedTabsChannel"
+    }
+
+    // Use an incrementing notification ID since they have the same tag.
+    var sendTabNotificationIdCount = 0
+
+    init {
+        // Create the notification channels we are going to use, but only on API 26+ because the NotificationChannel
+        // class is new and not in the support library.
+        // We pick 'IMPORTANCE_HIGH' priority because this is a user-triggered action that is
+        // expected to be part of a continuity flow. That is, user is expected to be waiting for
+        // this notification on their device; make it obvious.
+        if (SDK_INT >= Build.VERSION_CODES.O) {
+            createNotificationChannel(
+                RECEIVE_TABS_CHANNEL_ID,
+                AndroidNotificationManager.IMPORTANCE_HIGH,
+                context.getString(R.string.fxa_received_tab_channel_name),
+                context.getString(R.string.fxa_received_tab_channel_description)
+            )
+        }
+    }
+
+    private val logger = Logger("NotificationManager")
+    fun showReceivedTabs(device: Device?, tabs: List<TabData>) {
+        // In the future, experiment with displaying multiple tabs from the same device as as Notification Groups.
+        // For now, a single notification per tab received will suffice.
+        logger.debug("Showing ${tabs.size} tab(s) received from deviceID=${device?.id}")
+
+        tabs.forEach { tab ->
+            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(tab.url)).apply {
+                flags = Intent.FLAG_ACTIVITY_NEW_TASK
+            }
+            val pendingIntent: PendingIntent =
+                PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_ONE_SHOT)
```

Comment:
request code should be unique?

Reply:
Hmm.. I don't think we look at the request code any where right now, but I've added it because that seems fair to do!